web/documentserver-example/java-spring/src/main/java/com/onlyoffice/integration/documentserver/managers/history/DefaultHistoryManager.java

```diff
@@ -292,10 +293,13 @@ public String getHistoryData(final String fileName, final String version, final
 
                     // write information about previous file version to the data object
                     dataObj.put("previous", prevInfo);
-                    // write the path to the diff.zip archive with differences in this file version
-                    Integer verdiff = i - 1;
-                    dataObj.put("changesUrl", documentManager
-                            .getHistoryFileUrl(fileName, verdiff, "diff.zip", true));
+
+                    if (diffExists(histDir, i)) {
+                        // write the path to the diff.zip archive with differences in this file version
+                        Integer verdiff = i - 1;
+                        dataObj.put("changesUrl", documentManager
+                                .getHistoryFileUrl(fileName, verdiff, "diff.zip", true));
+                    }
```

Comment:
Maybe move `- 1` from function to argument in line 297? It will be clearer this way, i think. Also you can lift up `verdiff` or delete it.

Reply:
> Maybe move `- 1` from `diffExists` function to argument in line 297? It will be clearer this way, i think. Also you can lift up `verdiff` or delete it.

lifted verdiff variable